JoinTable.lua

```diff
@@ -8,11 +8,18 @@ function JoinTable:__init(dimension, nInputDims)
    self.nInputDims = nInputDims
 end
 
-function JoinTable:updateOutput(input)
+function JoinTable:_getPositiveDimension(input)
    local dimension = self.dimension
-   if self.nInputDims and input[1]:dim()==(self.nInputDims+1) then
-       dimension = dimension + 1
+   if dimension < 0 then
+      dimension = input:dim() + dimension + 1
```

Comment:
if the input is 4-dimensional, and i give dimension to be -1, then I'd expect it to join dim-3, but with this formulation, dimension will be 4. What am I missing here...